### PR TITLE
Compare individuals for equality

### PR DESCRIPTION
**Pull request by N-Wouda**
This PR adds an equality operator for individuals, so we can compare them to see if they're the same.

### CODE REVIEW

File: pyvrp/Population.py [hunk 1]
```diff
@@ -87,20 +87,31 @@ def add(self, individual: Individual):
...
+                if id(other) == id(individual):
                     del prox[idx]
                     break
 
         for item in self:  # remove individual from subpopulation.
-            if item[0] == individual:
+            if id(item[0]) == id(individual):
                 self._items.remove(item)
                 break
         else:
```
Comment (N-Wouda) [inline pyvrp/Population.py:114]: This part used to already be by identity (in CPython: memory location), but it's more important to be explicit about this now that we can also compare by equality using ==.

File: pyvrp/Population.py [hunk 2]
```diff
@@ -87,20 +87,31 @@ def add(self, individual: Individual):
...
         for _, _, prox in self:  # remove individual from other proximities
             for idx, (other, _) in enumerate(prox):
-                if other == individual:
+                if id(other) == id(individual):
                     del prox[idx]
                     break
 
         for item in self:  # remove individual from subpopulation.
-            if item[0] == individual:
+            if id(item[0]) == id(individual):
```
Comment (leonlan) [inline pyvrp/Population.py:111]: Question: Is `id(x) == id(y)` preferred over/different from `x is y`?

Comment (leonlan) [inline pyvrp/Population.py:111]: Suggestion: can we change `_Item` from tuple to a dataclass? Then we can be more explicit about what we're accessing from the items: `id(item.individual) == id(individual)`.

Comment (N-Wouda) [reply pyvrp/Population.py:111]: No, I momentarily forgot that existed!

Comment (N-Wouda) [reply pyvrp/Population.py:111]: I do want to keep indexing and unpacking around, so I'll go for a named tuple. That way we get the best of both worlds!

**Review by leonlan — APPROVED** [review]

**Pull request merged**